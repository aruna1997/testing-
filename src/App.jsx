import React,{Component} from 'react';

export const doincrement=(prevState)=>({
        counter:prevState.counter+1
}) 
export const dodecrement=(prevState)=>({
    counter:prevState.counter-1
})

export const ok=()=>{
    console.log('check');
    
}

class App extends Component
{
    constructor(props)
    {
        super(props);
        this.state={
            counter:0
        }
        this.increment=this.increment.bind(this);
        this.decrement=this.decrement.bind(this);
    }
    increment()
    {
        this.setState(doincrement);
    }
    decrement()
    {
        this.setState(dodecrement);
    }
    
    render()
    {
        const {counter}=this.state;
        return(
            <div>
                <h1>Counter App</h1>
                <p>{counter}</p>
                <button onClick={this.increment}>increment</button>
                <button onClick={this.decrement}>decrement</button>
            </div>
        )
    }
}
export default App;