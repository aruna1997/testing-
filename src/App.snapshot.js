import React from "react";
import renderer from "react-test-renderer";
import App from "./App.jsx";

describe('App snapshot',()=>{
    test('renders',()=>{
       const component=renderer.create(<App/>);
       let tree=component.toJSON();
       expect(tree).toMatchSnapshot();
     })
})