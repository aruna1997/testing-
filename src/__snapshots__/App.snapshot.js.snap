// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    increment
  </button>
  <button
    onClick={[Function]}
  >
    decrement
  </button>
</div>
`;
